#! /bin/sh /usr/share/dpatch/dpatch-run
## 14_kfreebsd_build_fix.dpatch by Brian Nelson <pyro@debian.org> by way
## of Petr Salinger <Petr.Salinger@t-systems.cz>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes FTBFS on GNU/kFreeBSD

@DPATCH@
diff -urNad qt4-x11-4.0.1~/src/corelib/global/qglobal.h qt4-x11-4.0.1/src/corelib/global/qglobal.h
--- qt4-x11-4.0.1~/src/corelib/global/qglobal.h	2005-08-16 11:41:07.000000000 +0300
+++ qt4-x11-4.0.1/src/corelib/global/qglobal.h	2005-12-13 18:19:47.000000000 +0200
@@ -100,7 +100,7 @@
 #  define Q_OS_ULTRIX
 #elif defined(sinix)
 #  define Q_OS_RELIANT
-#elif defined(__linux__) || defined(__linux)
+#elif defined(__linux__) || defined(__linux) || defined(__GLIBC__)
 #  define Q_OS_LINUX
 #elif defined(__FreeBSD__) || defined(__DragonFly__)
 #  define Q_OS_FREEBSD
